apis/secretsmanager/v1alpha1/custom_types.go

```diff
@@ -20,10 +20,21 @@ import xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
 
 // CustomSecretParameters contains the additional fields for SecretParameters.
 type CustomSecretParameters struct {
-	// SecretRef is used to reference the secret and the key whose value you'd
-	// like to be stored in AWS.
-	// The value is stored as is with no change.
-	SecretRef xpv1.SecretKeySelector `json:"secretRef"`
+	// PayloadType lets you choose which part of the input secret should be used.
+	// SingleValue: Only value of the given key.
+	// Map: The whole content of the input secret will be used after converting
+	// it to JSON map.
+	// +kubebuilder:validation:Enum=SingleValue;Map
+	PayloadType string `json:"payloadType"`
```

Comment:
@muvaf instead of using the `crossplane-runtime` `SecretKeySelector`, can we just use a custom implementation where the `key` field is optional and if it is not provided then we use the whole secret map?

Reply:
That would be closer to the initial implementation where `key` was optional but in case `key` wasn't given and there was only one key, the value of that one was used. If there were more, only then a new JSON map would be constructed.

Anyway, the reason I ended up with a discriminator is that the shape of the content changes. For example, for the following secret:
```yaml
k8skey2:  value1
k8skey:   value2
```
`SingleKey` pointing to `k8skey` would send only `value2` (just like other secret inputs) but if you choose to send the whole secret a JSON map that includes the keys are sent (I'm describing this because I was confused initially, thinking in all cases we send a map with keys). The latter is also a valid use case because AWS Secret Manager support both key-value pairs as well as simple raw strings with the same mechanism.

So I can go with your suggestion, just wanted to make sure you know this is the case.

Reply:
Thanks for explaining this @muvaf :) If someone did want to send a single kv pair could they do something like the following with a `key` of `k8skey`?

```
k8skey: "smkey: smvalue"
```

Reply:
Yes, they are free to put anything they want there, including JSON map.